### DISCUSSION TITLE
Heaviside function

### DISCUSSION THREAD

**Discussion by s-celles** [JuliaControl/ControlSystems.jl#918] [General]
Hello,

I noticed looking at https://github.com/search?q=repo%3AJuliaControl%2FControlSystems.jl%20heaviside that heaviside function is not defined here.

Maybe it could make sense to have such a function into some of JuliaControl packages.

Use case can be step response of a simple SISO system (that's a teaching case) at a custom t0 time and with a custom U0 amplitude.

Related issue https://github.com/JuliaLang/julia/issues/14981

Matlab have such a function https://fr.mathworks.com/help/symbolic/heaviside.html
Numpy also https://numpy.org/doc/stable/reference/generated/numpy.heaviside.html

in `step` function it seems to be
https://github.com/JuliaControl/ControlSystems.jl/blob/0d825400f1fd0126a4ccaa63408ba2b38351d47a/src/timeresp.jl#L223

Some others classical input signals might also be implemented.

Kind regards

**Comment by baggepinnen** [comment]
The Heaviside function $h(t)$ in julia is `t > 0` (which is actually significantly shorter to write than `heaviside(t)`, or if you want the step to occur at another time $t_0$, `t >= t0`, or with a different amplitude, `A*(t >= t0)`.

I've been asked about simple input functions before, but I always concluded that they aren't worth the time and effort to implement and document since
- It's so incredibly easy to define them yourself
- Defining the function yourself lends itself very well to slight modifications and extensions
- Hiding a function behind a name like "Heaviside" makes it hard to find for a user who searches for "step function". It's not enough to have a function defined, a user must also know that it's there and what it is called. 
- The heaviside function in particular is more of a mathematical tool than something useful for computations, in fact, the discontinuity of this function makes numerical integration routines struggle and take small steps etc. The proper way to integrate such functions depends on the use case. Providing such a function encourages bad practices when it comes to numerical code.
